sdk/keyvault/Azure.Security.KeyVault.Administration/tests/AdministrationTestBase.cs

```diff
@@ -94,6 +94,13 @@ public sealed override void StartTestRecording()
         {
             base.StartTestRecording();
 
+            // If the AZURE_MANAGEDHSM_URL variable is not defined, we didn't provision one
+            // due to limitations: https://github.com/Azure/azure-sdk-for-net/issues/16531
+            if (string.IsNullOrEmpty(TestEnvironment.ManagedHsmUrl))
+            {
+                throw new IgnoreException($"Required variable 'AZURE_MANAGEDHSM_URL' is not defined");
```

Comment:
What if we throw in the ManagedHsmUrl getter?

```
=> GetRecordedOptionalVariable("AZURE_MANAGEDHSM_URL") ?? throw new IgnoreException($"Required variable 'AZURE_MANAGEDHSM_URL' is not defined");
```

Can we avoid all extra method altogether?

Reply:
Most of the other changes were either clean up (not setting `Client` again when we didn't need to - that was left-overs from some previous refactoring, it seems) or necessary to avoid empty recordings. Sure, it'd remove a couple lines of `if` and brackets, but ultimately doesn't change much. I could, in `KeysTestBase`, do that instead of having a localhost handy.

Reply:
The benefit would be not having to put this code in every class, your test would automatically be skipped as soon as you use something that's not available.

Reply:
Up to you.